actionlib_lisp/src/new_implementation/comm-state-machine.lisp

```diff
@@ -146,30 +142,31 @@
    transition-callback. If the result was processed before the 
    last status update the transition-callback gets called even
    if the state-machine doesn't change"
-  (if (and (or (eql (name (get-current-state (stm csm))) :done)
-               (process-signal (stm csm) signal))
-           (transition-cb csm))
-      (funcall (transition-cb csm))))
+  (when (and (or (eql (name (get-current-state (stm csm))) :done)
```

Comment:
@jannikb I am repeatedly stumbling over this piece of code. I am just pointing this out as a point where we could improve readability.. I am aware that it was almost like this in the old version.